router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go

```diff
@@ -1,403 +1,355 @@
-		conf:              conf,
-		logger:            logger.Child("Marketo").Child("MarketoBulkUploader"),
-		statsFactory:      statsFactory,
-		timeout:           conf.GetDuration("HttpClient.marketoBulkUpload.timeout", 30, time.Second),
+type MarketoAsyncFailedInput struct {
+	Message  map[string]interface{}
+	Metadata struct {
+		JobID int64
 	}
-	return marketoBulkUpload, nil
 }
 
-type marketoPollInputStruct struct {
-	ImportId   string                 `json:"importId"`
-	DestType   string                 `json:"destType"`
-	DestConfig map[string]interface{} `json:"config"`
+type MarketoAsyncFailedPayload struct {
+	Config   map[string]interface{}
+	Input    []MarketoAsyncFailedInput
+	DestType string
+	ImportId string
+	MetaData common.MetaDataT
 }
 
-func (b *MarketoBulkUploader) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
-	finalPollInput := marketoPollInputStruct{
-		ImportId:   pollInput.ImportId,
-		DestType:   "MARKETO_BULK_UPLOAD",
-		DestConfig: b.destinationConfig,
-	}
-	payload, err := json.Marshal(finalPollInput)
+const (
+	MARKETO_WARNING_HEADER = "Import Warning Reason"
+	MARKETO_FAILED_HEADER  = "Import Failure Reason"
+)
+
+func getImportingParameters(importID string) json.RawMessage {
+	return json.RawMessage(`{"importId": "` + importID + `"}`)
+}
+
+func (b *MarketoBulkUploader) Upload(asyncDestStruct *common.AsyncDestinationStruct) common.AsyncUploadOutput {
+	destination := asyncDestStruct.Destination
+	destinationID := destination.ID
+	filePath := asyncDestStruct.FileName
+	destType := destination.DestinationDefinition.Name
+	failedJobIDs := asyncDestStruct.FailedJobIDs
+	importingJobIDs := asyncDestStruct.ImportingJobIDs
+
+	input, err := readJobsFromFile(filePath)
 	if err != nil {
-		b.logger.Errorf("Error in Marshalling Poll Input: %v", err)
-		return common.PollStatusResponse{
-			StatusCode: 500,
-			HasFailed:  true,
+		return common.AsyncUploadOutput{
+			FailedJobIDs:  append(failedJobIDs, importingJobIDs...),
+			FailedReason:  err.Error(),
+			FailedCount:   len(failedJobIDs) + len(importingJobIDs),
+			DestinationID: destinationID,
 		}
 	}
 
-	pollURL, err := url.JoinPath(b.transformUrl, b.pollUrl)
+	payloadSizeStat := b.statsFactory.NewTaggedStat("payload_size", stats.GaugeType, map[string]string{
+		"module":   "batch_router",
+		"destType": destType,
+		"destID":   destinationID,
+	})
+	csvFilePath, headerRowOrder, insertedJobIDs, overflowedJobIDs, err := createCSVFile(destinationID, b.destinationConfig, input, b.dataHashToJobId)
+
+	b.logger.Infof("Number of jobs in the batch: %d", len(insertedJobIDs))
+
 	if err != nil {
-		b.logger.Errorf("Error in preparing poll url: %v", err)
-		return common.PollStatusResponse{
-			StatusCode: 500,
-			HasFailed:  true,
-			Error:      err.Error(),
+		return common.AsyncUploadOutput{
+			FailedJobIDs:  append(failedJobIDs, importingJobIDs...),
+			FailedReason:  "BRT: Error in Creating CSV File: " + err.Error(),
```

Comment:
```suggestion
			FailedReason:  fmt.Sprintf("BRT: Error in Creating CSV File: %v", err),
```

Reply:
We have a lot of unused functions in common.go , we should probably get rid of them

Reply:
Fixed